tsdb/engine/tsm1/compact.go

```diff
@@ -1064,11 +1064,11 @@ func (c *Compactor) writeNewFiles(generation, sequence int, src []string, iter K
 
 		// We've hit the max file limit and there is more to write.  Create a new file
 		// and continue.
-		if err == errMaxFileExceeded || err == ErrMaxBlocksExceeded {
+		if errors.Is(err, errMaxFileExceeded) || errors.Is(err, ErrMaxBlocksExceeded) {
```

Comment:
Not sure if it matters but since you are joining errors and then checking with `errors.Is` if there are multiple errors joined and `errMaxFileExceed` or `ErrMaxBlocksExceeded` is in the errors this will pass the test.

https://goplay.tools/snippet/mDQiVeDgHB3

I'm sure thats what you were expecting but just something I noticed with `errors.Is`

Reply:
Although the original code discards file closing errors on  `errMaxFileExceededor` or `ErrMaxBlocksExceeded`, behavior that using `errors.Is` essentially replicates because of the feature you mention above, on reflection I think we should report and abort compaction on file closing errors, which will require an additional commit to this PR.